src/teachy_tv.c

```diff
@@ -0,0 +1,1797 @@
+        .baseTile = 0x0,
+    },
+    {
+        .bg = 0x1,
+        .charBaseIndex = 0x0,
+        .mapBaseIndex = 0x1E,
+        .screenSize = 0x0,
+        .paletteMode = 0x0,
+        .priority = 0x0,
+        .baseTile = 0x0,
+    },
+    {
+        .bg = 0x2,
+        .charBaseIndex = 0x0,
+        .mapBaseIndex = 0x1D,
+        .screenSize = 0x0,
+        .paletteMode = 0x0,
+        .priority = 0x2,
+        .baseTile = 0x0,
+    },
+    {
+        .bg = 0x3,
+        .charBaseIndex = 0x2,
+        .mapBaseIndex = 0x1C,
+        .screenSize = 0x0,
+        .paletteMode = 0x0,
+        .priority = 0x3,
+        .baseTile = 0x0,
+    },
+};
+
+const struct WindowTemplate gTeachyTvWindowTemplateArray[] = 
+{
+    {
+        .bg = 0x1,
+        .tilemapLeft = 0x2,
+        .tilemapTop = 0xF,
+        .width = 0x1A,
+        .height = 0x4,
+        .paletteNum = 0x3,
+        .baseBlock = 0xEA,
+    },
+    {
+        .bg = 0x0,
+        .tilemapLeft = 0x4,
+        .tilemapTop = 0x1,
+        .width = 0x16,
+        .height = 0xC,
+        .paletteNum = 0x3,
+        .baseBlock = 0x152,
+    },
+    {
+        .bg = 0xFF,
+        .tilemapLeft = 0x0,
+        .tilemapTop = 0x0,
+        .width = 0x0,
+        .height = 0x0,
+        .paletteNum = 0x0,
+        .baseBlock = 0x0,
+    },
+};
+
+const struct ListMenuItem gTeachyTvListMenuItemArray[] = 
+{
+    {
+        .unk_00 = gUnknown_841B7A4,
+        .unk_04 = 0,
+    },
+    {
+        .unk_00 = gUnknown_841B7BC,
+        .unk_04 = 1,
+    },
+    {
+        .unk_00 = gUnknown_841B7D6,
+        .unk_04 = 2,
+    },
+    {
+        .unk_00 = gUnknown_841B7EE,
+        .unk_04 = 3,
+    },
+    {
+        .unk_00 = gUnknown_841B807,
+        .unk_04 = 4,
+    },
+    {
+        .unk_00 = gUnknown_841B81B,
+        .unk_04 = 5,
+    },
+
+    {
+        .unk_00 = gUnknown_841B836,
+        .unk_04 = -2,
+    },
+};
+
+const struct ListMenuItem gTeachyTvListMenuItemArray2[] = 
+{
+    {
+        .unk_00 = gUnknown_841B7A4,
+        .unk_04 = 0,
+    },
+    {
+        .unk_00 = gUnknown_841B7BC,
+        .unk_04 = 1,
+    },
+    {
+        .unk_00 = gUnknown_841B7D6,
+        .unk_04 = 2,
+    },
+    {
+        .unk_00 = gUnknown_841B7EE,
+        .unk_04 = 3,
+    },
+    {
+        .unk_00 = gUnknown_841B836,
+        .unk_04 = -2,
+    },
+};
+
+const struct ListMenuTemplate gTeachyTvListMenuTemplateArray = 
+{
+    .items = gTeachyTvListMenuItemArray,
+    .moveCursorFunc = NULL,
+    .itemPrintFunc = NULL,
+    .totalItems = 0x7,
+    .maxShowed = 0x6,
+    .windowId = 0x0,
+    .header_X = 0x0,
+    .item_X = 0x8,
+    .cursor_X = 0x0,
+    .upText_Y = 0x6,
+    .cursorPal = 0x1,
+    .fillValue = 0x0,
+    .cursorShadowPal = 0x2,
+    .lettersSpacing = 0x0,
+    .itemVerticalPadding = 0x0,
+    .scrollMultiple = 0x1,
+    .fontId = 0x2,
+    .cursorKind = 0x0,
+};
+
+const struct ScrollIndicatorArrowPairTemplate gTeachyTvScrollIndicatorArrowPairTemplateArray = 
+{
```

Comment:
TODO: Update the ScrollIndicatorArrowPairTemplate struct definition. Emerald's is replicated below. How sure are we that field 0e (`palNum` in emerald) is to be encoded by nybbles?
```c
struct ScrollArrowsTemplate
{
    u8 firstArrowType;
    u8 firstX;
    u8 firstY;
    u8 secondArrowType;
    u8 secondX;
    u8 secondY;
    u16 fullyUpThreshold;
    u16 fullyDownThreshold;
    u16 tileTag;
    u16 palTag;
    u8 palNum;
};
```